cg/apps/sequencing_metrics_parser/api.py

```diff
@@ -33,13 +33,44 @@ def create_undetermined_non_pooled_metrics(
     non_pooled_lanes_and_samples: list[tuple[int, str]] = (
         flow_cell.sample_sheet.get_non_pooled_lanes_and_samples()
     )
+    metrics_parser = BclConvertMetricsParser(flow_cell.path)
+    undetermined_metrics: list[SampleLaneSequencingMetrics] = []
+
+    for lane, sample_internal_id in non_pooled_lanes_and_samples:
+        if not metrics_parser.has_undetermined_reads_in_lane(lane):
+            continue
 
-    if flow_cell.bcl_converter == BclConverter.BCL2FASTQ:
-        return create_bcl2fastq_undetermined_metrics(
-            bcl2fastq_flow_cell_path=flow_cell.path,
-            non_pooled_lane_sample_pairs=non_pooled_lanes_and_samples,
+        # Passing Undetermined as the sample id is required to extract the undetermined reads data.
+        # BclConvert tags undetermined reads in a lane with the sample id "Undetermined".
+        metrics: SampleLaneSequencingMetrics = _create_bcl_convert_sequencing_metrics(
+            sample_internal_id=UNDETERMINED, lane=lane, metrics_parser=metrics_parser
         )
-    return create_bcl_convert_undetermined_metrics(
-        flow_cell_dir=flow_cell.path,
-        non_pooled_lane_sample_pairs=non_pooled_lanes_and_samples,
+        metrics.sample_internal_id = sample_internal_id
+        undetermined_metrics.append(metrics)
+    return undetermined_metrics
+
+
+def _create_bcl_convert_sequencing_metrics(
```

Comment:
moved this function here from `cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py`